File: GXpert/GXpert.Web/Modules/Institute/Institute/InstituteColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Institute.Columns;

[ColumnsScript("Institute.Institute")]
[BasedOnRow(typeof(InstituteRow), CheckNames = true)]
public class InstituteColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Description { get; set; }
    public short EType { get; set; }
    public string Address { get; set; }
    public string City { get; set; }
    public string StateTitle { get; set; }
    public string DistrictTitle { get; set; }
    public string TalukaTitle { get; set; }
    public string LocationInfo { get; set; }
    public DateTime EstablishmentDate { get; set; }
   
    public string PrincipalName { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Institute/InstituteColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { EInstituteType } from "../Web/Enums.EInstituteType";
import { InstituteRow } from "./InstituteRow";

export interface InstituteColumns {
    Id: Column<InstituteRow>;
    Name: Column<InstituteRow>;
    Description: Column<InstituteRow>;
    EType: Column<InstituteRow>;
    Address: Column<InstituteRow>;
    City: Column<InstituteRow>;
    StateTitle: Column<InstituteRow>;
    DistrictTitle: Column<InstituteRow>;
    TalukaTitle: Column<InstituteRow>;
    LocationInfo: Column<InstituteRow>;
    EstablishmentDate: Column<InstituteRow>;
    PrincipalName: Column<InstituteRow>;
}

export class InstituteColumns extends ColumnsBase<InstituteRow> {
    static readonly columnsKey = 'Institute.Institute';
    static readonly Fields = fieldsProxy<InstituteColumns>();
}

[EInstituteType]; // referenced types